docs/rfc/active/react-app/README.md

```diff
@@ -0,0 +1,158 @@
+    - Theme: How it looks (color, ux, assets, copy)
+    - Features: How it behaves (enable / disable features)
+      
+      out of the box!
+
+
+- **Coupling with GMA**: GMA concepts of [entity](https://github.com/linkedin/datahub/blob/master/datahub-web/@datahub/data-models/addon/entity/base-entity.ts) and [aspect](https://github.com/linkedin/datahub/blob/master/datahub-web/@datahub/data-models/addon/entity/utils/aspects.ts) are rooted in the Ember client. With the new client, we can revisit the abstractions exposed to the client side to simplify the client.
+
+A clean slate will allow us to address these items, improving the frontend development experience & making community contribution easier.
+
+It is important to note that we are not proposing deprecation of the Ember client at this time. Maintenance and feature development should be free to continue on Ember as the React app evolves in isolation.
+
+### Design Principles
+
+In developing the new application, it is important that we have an agreed-upon set of design principles to guide our decisions.
+
+Such principles should promote the health of the community (eg. by increasing the likelihood of contribution) & the value proposition of the DataHub product for organizations (eg. by permitting domain-specific modification of the default deployment).
+
+Specifically, the new client be
+
+1. **Extensible**
+- Modular, composable architecture 
+- Formal guidance on extending the client to support domain-specific needs
+- Easy to contribute!
+
+2. **Customizable**
+- Clear, consistent, & documented levers to alter style & behavior between DataHub deployments
+- Support injection of custom ‘applets’ or ‘widgets’ where appropriate
+
+3. **Scalable**
+- An architecture suited for scale, both in terms of additional metadata entities & novel features / use cases.
+
+These principles should serve as evaluation criteria by authors & reviewers of application changes.
+
+
+### Functional Requirements
+
+#### Near term
+
+Initially, our goal is to achieve functional parity with the existing Ember frontend for common use-cases. Specifically, the React app should support
+
+- Authenticating a user
+- Displaying metadata entities
+- Updating metadata entities  
+- Browsing metadata entities
+- Searching metadata entities
+- Managing a user account
+
+The finer details of which entities fall into each feature bucket will be dictated by the needs of the community, with the short-term milestone to achieve parity with entities appearing in the Ember client (Datasets, CorpUsers).
+
+
+#### Long term
+
+In the longer term, we will work with the community to define a more extensive functional road map, which may include
+
+- Providing migration pathway from the Ember application to the React application
```

Comment:
I would ask that this be done before any code commits. Ditto with GQL, tbh.

Until we have migration plans we'd need to maintain the old world too. So if we have two options for the UI (react, ember) and two for the midtier (GQL, legacy restli), then suddenly we also have 4 combinations of things to support, and the complexity of the software greatly increases.

Thinking about EoL for the things you want to replace should be done ahead of time. You might find it isn't as straightforward or feasible as you thought, and imo that's a pretty big cost to adding the new thing as a result.

Reply:
I guess if we also say "ember only supports legacy midtier; react only GQL" then its only two states, but might make it more difficult to get a deployment correct.

Reply:
Hmm maybe this isn't as big of a deal if the stacks are separate, you just flip a switch at some point and start serving the other app I guess. I think the only concern is getting apollo deployed, but that should be straight forward?

I guess I'm more ok with this being delayed if we go with this approach. Resolving. lol.

Reply:
Yes my understanding of this proposal is that the stacks are separate as you describe it. @jjoyce0510 to confirm.

Reply:
Yep, its later down in the RFC. Which is why I resolved this as I read more :p

Reply:
Your understanding is correct! 

"Getting apollo deployed" --> Not following this piece. If you mean getting a GraphQL endpoint at the frontend layer deployed, then yes it's true that will need to exist